tiledb/api/c_api/array_schema/array_schema_api_experimental.h

```diff
@@ -43,6 +43,34 @@
 extern "C" {
 #endif
 
+/**
+ * Creates a TileDB array schema object with specified creation time.
+ *
+ * Note: This is an experimental API for expert users only. This should be used
+ * with caution only as needed. Setting custom timestamps in general increases
+ * the likelihood of misuse and potential data corruption. This API intended to
+ * be used for handling schema evolution at custom timestamps.
```

Comment:
I would prefer a message that specifically discourages new use cases from using this API. Something like this:

```suggestion
 * Note: This API is provided for compatibity with existing workloads that ascribe
 * a semantic value to timestamps. Setting custom timestamps in general increases
 * the likelihood of misuse and potential data corruption. This API intended to
 * be used for handling schema evolution at custom timestamps and should not be used
 * by new workloads.
```